### DISCUSSION TITLE
Token importance for text classification - explainability with spaCy

### DISCUSSION THREAD

**Discussion by kormilitzin** [explosion/spaCy#9826] [Help: Coding & Implementations] [answered]
Hi all, I'm wondering if there is a common approach to shed light on the model explainability, for example, for a text classification task. Perhaps something similar to SHAP or Integrated Gradients? There have been a few attempts to get the token importance (e.g. [here](https://stackoverflow.com/questions/68545128/how-to-use-shap-with-spacy-models)), but I couldn't find consensus on this topic.

(I saw alibi, but this is not directly a spaCy-based method)

How can we interpret and explain predictions of spaCy models for text classification for example?

King regards

**Comment by polm** [comment, marked as answer]
We don't have a standard way to do this, so you'll need to look at third-party solutions like alibi for the time being. You might also want to take a look at work from Marco Tulio Ribeiro's research group, like [Anchor](https://github.com/marcotcr/anchor), which do seem intended for use with spaCy (if only very loosely integrated). (Note that based on the year the repo was released, I assume it's designed to work with spaCy v2, but since the integration is pretty loose it should be easy to adjust.)

**Reply by kormilitzin** [reply]
Thanks a lot @polm, that's helpful. I'll give a try to Anchor, but hope there will be more tight integration in the future.
